src/components/Tree/types.ts

```diff
@@ -0,0 +1,34 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { ReactElement, ReactNode } from 'react';
+
+import { OnDropCallback } from '@components/DropZone';
+
+export type TreeProps = {
+    id: string;
+    draggable?: boolean;
+    children: ReactElement<TreeItemProps> | Array<ReactElement<TreeItemProps>>;
+};
+
+type ContentComponentArguments = {
+    selected: boolean;
+    hovered: boolean;
+};
+
+type TreeItemWIthLabelProps = {
+    label?: string;
+    contentComponent?: never;
+};
+
+type TreeItemWIthContentComponentProps = {
+    label?: never;
```

Comment:
```suggestion
```

Reply:
Needs to be defined always as either `label` should used or `contentComponent`